templates/default/apache_site.conf.erb

```diff
@@ -51,8 +52,9 @@
 
     AllowOverride <%= @params['parse_htaccess'] ? "All" : "None" %>
 
-    Order allow,deny
-<%= @params['allow_from'] ? @params['allow_from'].map { |ip| "    Allow from #{ip}\n" } : "    Allow from all" %>
+<%= @params['allow_from'] || @params['basic_username'] ? "    Order deny,allow" : "    Order allow,deny" %>
+<%= @params['allow_from'] ? @params['allow_from'].map { |ip| "    Allow from #{ip}" }.join("\n") : '' %>
+<%= @params['allow_from'] || @params['basic_username'] ? "    Deny from all" : "    Allow from all" %>
```

Comment:
It doesn't look necessary to `Order deny,allow .... Deny from all` as for `Order allow,deny` implies Deny from all if no allow match. The older way therefore is cleaner if it is just fixed by itself:

```
Order allow,deny
<%= @params['allow_from'] ? @params['allow_from'].map { |ip| "    Allow from #{ip}" }.join("\n") : '    Allow from all' %>
```